<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LauncherWindow</class>
 <widget class="QMainWindow" name="LauncherWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Haptic Interface Launcher</string>
  </property>
  <property name="styleSheet">
   <string>QMainWindow {
    background-color: #f5f5f5;
}

QPushButton {
    background-color: #4CAF50;
    color: white;
    font-weight: bold;
    border: none;
    border-radius: 8px;
    padding: 15px;
    font-size: 14px;
    min-height: 40px;
}

QPushButton:hover {
    background-color: #45a049;
}

QPushButton:pressed {
    background-color: #3d8b40;
}

QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
}

QLabel#titleLabel {
    font-size: 24px;
    font-weight: bold;
    color: #333333;
    margin: 20px 0px;
}

QLabel#subtitleLabel {
    font-size: 14px;
    color: #666666;
    margin-bottom: 30px;
}

QGroupBox {
    font-weight: bold;
    border: 2px solid #cccccc;
    border-radius: 8px;
    margin-top: 10px;
    padding-top: 10px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 5px 0 5px;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>20</number>
    </property>
    <property name="leftMargin">
     <number>30</number>
    </property>
    <property name="topMargin">
     <number>30</number>
    </property>
    <property name="rightMargin">
     <number>30</number>
    </property>
    <property name="bottomMargin">
     <number>30</number>
    </property>
    <item>
     <widget class="QLabel" name="titleLabel">
      <property name="text">
       <string>Haptic Interface Control Center</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="subtitleLabel">
      <property name="text">
       <string>Connect your USB device and launch the desired interface module</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="connectionGroupBox">
      <property name="title">
       <string>Device Connection</string>
      </property>
      <layout class="QHBoxLayout" name="connectionLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <item>
        <widget class="QLabel" name="statusIndicator">
         <property name="text">
          <string>‚óè</string>
         </property>
         <property name="styleSheet">
          <string>color: #f44336; font-size: 16px; font-weight: bold;</string>
         </property>
         <property name="toolTip">
          <string>Connection Status</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="statusText">
         <property name="text">
          <string>Disconnected</string>
         </property>
         <property name="styleSheet">
          <string>color: #666; font-size: 12px; margin-left: 5px; margin-right: 10px;</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="connectionButton">
         <property name="text">
          <string>Connect USB Device</string>
         </property>
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>35</height>
          </size>
         </property>
         <property name="styleSheet">
          <string>QPushButton {
    background-color: #4CAF50;
    color: white;
    font-weight: bold;
    border: none;
    border-radius: 6px;
    padding: 8px 16px;
}
QPushButton:hover {
    background-color: #45a049;
}
QPushButton:pressed {
    background-color: #3d8b40;
}</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="modulesGroupBox">
      <property name="title">
       <string>Interface Modules</string>
      </property>
      <layout class="QVBoxLayout" name="modulesLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <item>
        <widget class="QPushButton" name="droneControlButton">
         <property name="text">
          <string>üöÅ Drone Control GUI</string>
         </property>
         <property name="toolTip">
          <string>Launch the main drone control interface with 3D view, 2D map, and actuator controls</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="waveformDesignerButton">
         <property name="text">
          <string>üåä Waveform Designer</string>
         </property>
         <property name="toolTip">
          <string>Launch the Universal Event Designer for creating custom haptic waveforms</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="patternGuiButton">
         <property name="text">
          <string>üìä Pattern GUI</string>
         </property>
         <property name="toolTip">
          <string>Launch the Haptic Pattern GUI for creating and testing vibration patterns</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>500</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>